Source/WebCore/platform/calc/CalculationValueMap.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ * Copyright (C) 2003-2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "CalculationValue.h"
+#include <wtf/HashMap.h>
+
+namespace WebCore {
+
+class CalculationValueMap {
+public:
+    static CalculationValueMap& calculationValues();
+
+    CalculationValueMap();
+
+    unsigned insert(Ref<CalculationValue>&&);
+    void ref(unsigned handle);
+    void deref(unsigned handle);
+
+    CalculationValue& get(unsigned handle) const;
+
+private:
+    struct Entry {
+        uint64_t referenceCountMinusOne { 0 };
+        RefPtr<CalculationValue> value;
+        Entry() = default;
+        Entry(Ref<CalculationValue>&&);
+    };
+
+    unsigned m_nextAvailableHandle { 1 };
+    HashMap<unsigned, Entry> m_map;
+};
+
+inline CalculationValueMap::Entry::Entry(Ref<CalculationValue>&& value)
+    : value(WTFMove(value))
+{
+}
+
+inline CalculationValueMap::CalculationValueMap()
+    : m_nextAvailableHandle(1)
+{
+}
+
```

Comment:
```suggestion
```

Reply:
I can also move it to cpp file, since the only caller is also in that cpp file.